### PR TITLE
Fix minimizer renaming functions

### PR DESCRIPTION
**Pull request by diwic**
When running the code through a minimizer (e g through Webpack/Terser),
the minimizer renames functions such as "$asBoolean". However, when the
function is called, it's hard coded as the string "$asBoolean", so the
generated code fails with "$asBoolean is not defined".

Make sure that the original $asBoolean function is included, and not
a renamed version of it, by making $asBoolean and friends string constants.

Fixes: https://github.com/fastify/fast-json-stringify/issues/338

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included

~~No new tests or benchmarks.~~ The one added test causes the time to run the tests to increase from 2 to 7 seconds.

Benchmarks seem to go up or down a few percent, probably within the margin of error.

- [x] documentation is changed or added

N/A

- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Could you add a test so that we do not regress?

**Comment by diwic** [comment]
@mcollina I'm a little hesitant to do so given that it does not really fit into the testing infrastructure. It would also require an additional devDependency on webpack and increase the time to run the tests, although I'm not certain how much.
Do you want me to try regardless?

Otherwise a comment explaining why things look the way they do, could be a reasonable compromise between testing time and regression risk?

**Comment by mcollina** [comment]
@jsumners @delvedor wdyt?

I'll prefer to have a testing strategy for this.

File: index.js [hunk 1]
```diff
@@ -928,7 +916,7 @@ function buildObject (location, code, name) {
   if (schema.nullable) {
     code += `
       if(input === null) {
-        return '${$asNull()}';
+        return 'null';
```
Comment (jsumners) [inline index.js:919]: Why this change?

File: index.js [hunk 2]
```diff
@@ -967,7 +955,7 @@ function buildArray (location, code, name, key = null) {
   if (schema.nullable) {
     code += `
       if(obj === null) {
-        return '${$asNull()}';
+        return 'null';
```
Comment (jsumners) [inline index.js:958]: Why this change?

File: index.js [hunk 3]
```diff
@@ -372,7 +359,7 @@ function $asStringSmall (str) {
       surrogateFound = true
     }
     if (point === 34 || point === 92) {
-      result += str.slice(last, i) + '\\'
+      result += str.slice(last, i) + '\\\\'
```
Comment (jsumners) [inline index.js:362]: Why this change?

**Comment by jsumners** [comment]
> @jsumners @delvedor wdyt?
> 
> I'll prefer to have a testing strategy for this.

If there isn't an accompanying test, I don't know what this is fixing.

File: index.js (hunk 1 above)
Comment (diwic) [reply index.js:919]: The $asNull function is now inside a string constant, so it is not available to be called in this context.

File: index.js (hunk 2 above)
Comment (diwic) [reply index.js:958]: Same here - the $asNull function is now inside a string constant, so it is not available to be called in this context.

File: index.js (hunk 3 above)
Comment (diwic) [reply index.js:362]: $asStringSmall is now inside an es6 template string. Inside those strings, slashes need to be escaped.

**Comment by diwic** [comment]
@mcollina  @jsumners  All right, I've added a test case now (in a separate commit).

File: index.js (hunk 1 above)
Comment (jsumners) [reply index.js:919]: Is it not the case that an alternative function can be provided for `$asNull` and that such an alternative could return a value other than `'null'`?

Comment (diwic) [reply index.js:919]: Not to my knowledge? If so I don't see how such a function would be provided.

Comment (diwic) [reply index.js:919]: I could change it to `return $asNull()` I guess, but then there are other places in the code which just return `'null'` anyway, so it would still make much difference in terms of consistency.

**Comment by jsumners** [comment]
I'm not super familiar with how all of this works. So I'll defer to @Eomm.

File: index.js (hunk 1 above)
Comment (Eomm) [reply index.js:919]: there is still one `$asNull` call. I would remove it and remove the function aswell

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by diwic** [comment]
Thanks everybody for the quick reviews! Much appreciated.
